src/main/scala/nl.knaw.dans.easy.deposit/servlets/DepositServlet.scala

```diff
@@ -128,13 +129,14 @@ class DepositServlet(app: EasyDepositApiApp)
       uuid <- getUUID
       path <- getPath
       _ <- isMultipart
-      newFileItems <- getFileItems
-        .toStream
-        .withFilter(_.name.blankOption.isDefined)
-        .map { fileItem => postFileItem(uuid, path, fileItem) }
-        .failFast
-        .map(_.withFilter(_._1).map(_._2))
-    } yield Ok() // TODO create multiple location headers from newFileItems?
+      fileItems = fileMultiParams.valuesIterator.flatten.buffered
+      _ = while (fileItems.head.name.trim.isEmpty) { fileItems.next() } // skip leading form fields without selected files
```

Comment:
you're probably looking for [`fileItems.dropWhile(_.name.trim.isEmpty)`](https://www.scala-lang.org/api/current/scala/collection/Seq.html#dropWhile(p:A=%3EBoolean):Repr) and then the rest of the old code.

Reply:
No, head of a buffered iterator is peeking without forwarding the pointer.

Reply:
I know, but `fileItems.next()` IS forwarding the pointer

Reply:
But it should only be forwarded if the condition applies, not before it is checked.